core/src/main/java/org/elasticsearch/action/admin/indices/stats/CommonStatsFlags.java

```diff
@@ -192,6 +208,9 @@ public void readFrom(StreamInput in) throws IOException {
         groups = in.readStringArray();
         fieldDataFields = in.readStringArray();
         completionDataFields = in.readStringArray();
+        if (in.getVersion().onOrAfter(Version.V_3_0_0)) {
+            detailedSegmentsStats = in.readBoolean();
+        }
```

Comment:
Maybe add an else clause here to set `detailedSegmentsStats` to `false` so it's explicit?